### ISSUE TITLE
fork() is not supported on windows

### ISSUE THREAD

**Issue by TheDruidsKeeper** [microsoftarchive/redis#60]
This is what I did to produce this:
1. Run a redis master
2. Run a redis slave connected to the master
3. Run the sentinel to monitor the master, sentinel is configured to run a script upon failover
4. Kill the redis master

Failover works, but you instantly start getting errors from the sentinel trying to execute the script: "-script-error [file name] 99 0"

This messages comes from src/sentinel.c:662

``` c
void sentinelRunPendingScripts(void) {
...
  pid = fork();

  if (pid == -1) {
    sentinelEvent(REDIS_WARNING,"-script-error", ...
```

Ultimately comes down to the fork() failing because it is not supported on windows subsystems.

**Comment by m-manu** [comment]
Cygwin has implemented `fork()` for Windows. With Cygwin/GCC, you can actually build http://github.com/antirez/redis with just 1-2 changes.

**Comment by clca** [comment]
we will fix the problem on the sentinel side as well. We are not going to use Cyqwin but 'mimic' the fork behavior like we are doing on the Redis server. Cygwin introduces a fairly high overhead that would make Redis performance a lot slower than the Linux counterpart

**Comment by jepickett** [comment]
This is fixed in 2.6.14. fork() was being used to run execev() asynchronously. I replaced this with a call to CreateProcess() since it is inherently asynchronous. I also made some changes in tracking child process termination so that it would work more reliably on windows.

**Comment by jepickett** [comment]
2.6.14 and 2.8.4 have been released. The signed binaries are available via NuGet and Chocolatey. The fork problems have been resolved with these versions.
